### PR TITLE
Jay level 5a

### PR DESCRIPTION
**Pull request by jayfriso**
Level 5a and 5b Implementation.
-Added sprites to Level 5a.
- Implemented Level 5b and added sprites.
- Implemented the multi colored doll (using animation override controller).

Level 5a: 
5a

Level 5b:
5b

### CODE REVIEW

File: Assets/Scripts/CameraFollow.cs [hunk 1]
```diff
@@ -6,14 +6,16 @@
...
+	public float initialSpeed = 8f;
 	public float xMargin = 1f; // Distance in the x axis the player can move before the camera follows.
     public float yMargin = 1f; // Distance in the y axis the player can move before the camera follows.
     public float xSmooth = 8f; // How smoothly the camera catches up with it's target movement in the x axis.
     public float ySmooth = 8f; // How smoothly the camera catches up with it's target movement in the y axis.
     public Vector2 maxXAndY; // The maximum x and y coordinates the camera can have.
     public Vector2 minXAndY; // The minimum x and y coordinates the camera can have.
 
+
+    public float currentDollSpeed = 8f;
```
Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs]: can you make this follow the doll manager?

File: Assets/Scripts/Manager/GhostSwitchManager.cs [hunk 2]
```diff
@@ -195,14 +197,14 @@ bool hasGhostWall (GameObject doll)
 
 	// switch control of the player to the doll, as well as update the environment
 	void possess (GameObject player, GameObject doll)
-	{
-		dollsUpdated = false;
+	{     
+        dollsUpdated = false;
```
Comment (jeaniewu) [inline Assets/Scripts/Manager/GhostSwitchManager.cs:200]: whats with the change of indentation here?

File: Assets/Scripts/receiveSignal/receiveSignalSawBladeSpeed.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class receiveSignalSawBladeSpeed : receiveSignal {
```
Comment (jeaniewu) [inline Assets/Scripts/receiveSignal/receiveSignalSawBladeSpeed.cs:4]: is this for changing the speed when you push a certain button? if so make sure you put the corresponding trello task on done

Comment (jayfriso) [inline Assets/Scripts/receiveSignal/receiveSignalSawBladeSpeed.cs:4]: Yes it is actually, but I haven't exactly added the part to level 4 which is what the task was about so It's not technically done. I put it in my in progress though.

File: Assets/Scripts/CameraFollow.cs [hunk 4]
```diff
@@ -6,15 +6,17 @@
...
 {
-	public float maxSpeed = 10f;
 	public float xMargin = 1f; // Distance in the x axis the player can move before the camera follows.
     public float yMargin = 1f; // Distance in the y axis the player can move before the camera follows.
     public float xSmooth = 8f; // How smoothly the camera catches up with it's target movement in the x axis.
     public float ySmooth = 8f; // How smoothly the camera catches up with it's target movement in the y axis.
     public Vector2 maxXAndY; // The maximum x and y coordinates the camera can have.
     public Vector2 minXAndY; // The minimum x and y coordinates the camera can have.
 
+    private float maxSpeed;
```
Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs]: sorry jay, I think I gave you a really bad advice, camerafollow should only care about the current player's max speed, it shouldn't do any of the calculation nor find out what the max speed is from the doll manager

File: Assets/Scripts/CameraFollow.cs [hunk 5]
```diff
@@ -23,7 +25,9 @@ private void Start()
     {
         // Setting up the reference.
 		findPlayer();
```
Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs]: I think you can just add in the findPlayer() method to update the current camera maxSpeed, using the player's maxSpeed (that might have been slowed down)

Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs:26]: I think you can just add in maxSpeed = player.maxSpeed in findPlayer() method, that way u get the latest maxSpeed - one that might have been slowed down in controller2

File: Assets/Scripts/CameraFollow.cs [hunk 6]
```diff
@@ -101,7 +105,7 @@ private void TrackPlayer()
 		float moveHorizontal = Input.GetAxis ("Horizontal");
 		float moveVertical = Input.GetAxis ("Vertical");
 
-		transform.position += new Vector3 (moveHorizontal, moveVertical, 0).normalized * Time.deltaTime * maxSpeed;
+		transform.position += new Vector3 (moveHorizontal, moveVertical, 0).normalized * Time.deltaTime * currentDollSpeed;
```
Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs:107]: here can just use maxSpeed

File: Assets/Scripts/CameraFollow.cs [hunk 7]
```diff
@@ -125,5 +129,9 @@ public void zoomOut()
 
         }
     }
+
```
Comment (jeaniewu) [inline Assets/Scripts/CameraFollow.cs]: don't need this method anymore

File: Assets/Scripts/Manager/GhostSwitchManager.cs [hunk 8]
```diff
@@ -126,7 +126,9 @@ public void updateDolls ()
 	// make the camera follow the current player
 	void updateCamera ()
 	{
-		Camera.main.GetComponent<CameraFollow> ().findPlayer ();
+        CameraFollow cameraFollow = Camera.main.GetComponent<CameraFollow>();
+        cameraFollow.findPlayer ();
+        cameraFollow.setCurrentDollSpeed(player.GetComponent<Controller2>().maxSpeedSlowerBy);
 	}
```
Comment (jeaniewu) [inline Assets/Scripts/Manager/GhostSwitchManager.cs:131]: if you make the changes I suggested, here you can just findPlayer() as you assume that the method take care of changing the speed on its own

**Comment by jeaniewu** [comment]
sorry @jayfriso I think I mislead you haha I posted new comments

**Comment by jayfriso** [comment]
@jeaniewu Ok I will fix by making a variable of dollSpeed in each doll's controller2. Then the camera will just grab this speed.

**Pull request merged**